contrib/dockerd-rootless-setuptool.sh

```diff
@@ -228,8 +228,8 @@ init() {
 		fi
 	fi
 
-	# instructions: validate subuid/subgid files for current user
-	if ! grep -q "^$USERNAME_ESCAPED:\|^$(id -u):" /etc/subuid 2> /dev/null; then
+	# instructions: validate subuid/subgid for current user
+	if ! getsubids "$USERNAME" > /dev/null 2>&1 && ! getsubids "$(id -u)" > /dev/null 2>&1; then
```

Comment:
This command is not present on old distro

Reply:
I was curious about that indeed (if it's always available). Do you know what distros don't  provide it? (or is this old, like CentOS 7 / RHEL 7)?

Reply:
Silly idea; I  think when this script runs, we do already have  `dockerd` present, right? Would this be something `dockerd` would validate (we do have `dockerd --validate` to validate the configuration), and we could execute `dockerd` with some options set to do this?

Reply:
NSSwitch support is included in `shadow` 4.9 and `getsubids` in 4.10, which is very close.

However, a quick search shows that distros as new as Ubuntu 22.04 and Debian 11 didn’t come up with these two changes and stayed with 4.8. I’ll try to tweak the change to add a fallback if the tool is unavailable. Hopefully no one is using 4.9 which the current solution will still give a false alarm.

Reply:
The `dockerd` approach sounds appealing but modifying it is beyond my knowledge of Moby. I’m still going to finalize in the current situation, and maybe someday we will be able to do all these pre-flight checks in `dockerd`.